helm/external-dns-app/templates/_helpers.tpl

```diff
@@ -22,5 +22,48 @@ helm.sh/chart: {{ include "chart" . | quote }}
 Selector labels
 */}}
 {{- define "labels.selector" -}}
-app: {{ .Values.name | quote }}
+app: {{ .Release.Name | quote }}
 {{- end -}}
+
+{{/*
+Create the list of domains to update
+*/}}
+{{- define "domain.list" -}}
+{{- if .Values.externalDNS.domainFilterList -}}
+{{- range .Values.externalDNS.domainFilterList }}
+{{ printf "- --domain-filter=%s" . }}
+{{- end }}
+{{- else }}
+{{- if eq .Values.aws.access "external" }}
+{{- printf "- --domain-filter=%s" .Values.aws.baseDomain }}
+{{- else }}
+{{- printf "- --domain-filter=%s" .Values.baseDomain }}
+{{- end }}
+{{- end }}
+{{- end }}
```

Comment:
we need to accommodate multiple scenarios here. if the customer provides a list of domains to update then we use that, if not then we take the value of `aws.baseDomain` (user must provide this), and finally we default to `baseDomain` - this is provided by the cluster values when installing from our default catalog, so this allows this to still work fine as a default preinstalled app.

Reply:
Do we also need to take care of other providers like `azure`?

Reply:
no, it seems azure config is done entirely by providing a config file from a hostpath on a node's disk - the chart just assumes the file is there if you deploy on azure.